heroic-elasticsearch-utils/src/test/java/com/spotify/heroic/elasticsearch/ScrollTransformTest.java

```diff
@@ -0,0 +1,167 @@
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
+import org.mockito.stubbing.OngoingStubbing;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ScrollTransformTest {
+    @Mock
+    private AsyncFramework async;
+
+    @Mock
+    private AsyncFuture<LimitedSet> resolved;
+
+    @Mock
+    Supplier<AsyncFuture<SearchResponse>> scroller;
+
+    @Mock
+    AsyncFuture<SearchResponse> response;
+
+    @Mock
+    SearchResponse searchResponse;
+
+    @Mock
+    SearchHits searchHits;
+
+    private final SearchHit[] searchHits1 = {
+        mock(SearchHit.class),
+        mock(SearchHit.class),
+        mock(SearchHit.class),
+    };
+
+    private final SearchHit[] searchHits2 = {
+        mock(SearchHit.class),
+        mock(SearchHit.class),
+        mock(SearchHit.class),
+    };
+
+    private final SearchHit[] emptySearchHits = {};
+
+    @Before
+    public void setup() {
+        doReturn(resolved).when(async).resolved(any(LimitedSet.class));
+        doReturn(searchHits).when(searchResponse).getHits();
+
+        doReturn(response).when(scroller).get();
+        doAnswer(new Answer<AsyncFuture<LimitedSet<Integer>>>() {
+            public AsyncFuture<LimitedSet<Integer>> answer(
+                InvocationOnMock invocation
+            ) throws Exception {
+                LazyTransform<SearchResponse, LimitedSet<Integer>> transform =
+                    (LazyTransform<SearchResponse, LimitedSet<Integer>>)
+                        invocation.getArguments()[0];
+                return transform.transform(searchResponse);
+            }
+        }).when(response).lazyTransform(any(LazyTransform.class));
+    }
+
+    public void setSearchHitPages(SearchHit[]... pages) {
+        OngoingStubbing<SearchHit[]> stub = when(searchHits.getHits());
+        for (final SearchHit[] page : pages) {
+            stub = stub.thenReturn(page);
+        }
+    }
+
+    public ScrollTransform<Integer> createScrollTransform(
+        Integer limit,
+        Supplier<AsyncFuture<SearchResponse>> scroller
+    ){
+        final OptionalLimit optionalLimit = OptionalLimit.of(limit);
+        return new ScrollTransform<>(async, optionalLimit, scroller, Object::hashCode);
```

Comment:
[Object#hashCode()](https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#hashCode--) is not guaranteed to be unique in the local runtime environment. It would be safer to use `Function.identity()` and do equals comparisons on the actual SearchHits. These should be unique since they are individual mocks.